nautobot_floor_plan/models.py

```diff
@@ -145,8 +184,36 @@ def clean(self):
         - Ensure that the Rack if any belongs to the correct Location.
         - Ensure that this FloorPlanTile doesn't overlap with any other FloorPlanTile in this FloorPlan.
         """
-        # x <= 0, y <= 0 are covered by the base field definitions
         super().clean()
+        FloorPlanTile.allocation_type_assignment(self)
+
+        def group_tile_bounds(rack):
+            """Validate the overlapping of group tiles."""
+            if rack is not None:
+                if x_max > ox_max or x_min < ox_min:
+                    raise ValidationError(
+                        {f"Rack {self.rack} must not extend beyond the boundary of the defined group tiles"}
+                    )
+                if y_max > oy_max or y_min < oy_min:
+                    raise ValidationError(
+                        {f"Rack {self.rack} must not extend beyond the boundary of the defined group tiles"}
+                    )
+                self.on_group_tile = True
+                if orack_group is not None:
+                    if orack_group != rack_group:
+                        # If rack is not in the correct rack group or rack with with a rack group is being placed on a status group tile?
+                        raise ValidationError({"rack_group": f"Rack {self.rack} must belong to {orack_group}"})
+            if rack is None:
+                if x_max > ox_max or x_min < ox_min:
+                    # Allow for the extending of exising underlying Rack Group and Status tiles
+                    if orack_group is not None and allocation_type != AllocationTypeChoices.RACKGROUP:
+                        raise ValidationError({"Tile must not extend beyond the boundary of the defined group tiles"})
+                if y_max > oy_max or y_min < oy_min:
+                    # Allow for the extending of exising underlying Rack Group and Status tiles
+                    if orack_group is not None and allocation_type != AllocationTypeChoices.RACKGROUP:
+                        raise ValidationError({"Tile must not extend beyond the boundary of the defined group tiles"})
```

Comment:
Not sure I understand the validation case that's being checked for here. Can you elaborate in the comment?

Reply:
Changing comment to
# RACKGROUP tiles can grow and shrink but not overlap other RACKGROUP tiles